From 4101abafc359741f90d29f85fc9481354ad3f163 Mon Sep 17 00:00:00 2001
From: Tomasz Duszynski <tomasz.duszynski@cavium.com>
Date: Thu, 8 Nov 2018 09:41:35 +0100
Subject: [PATCH 0753/1051] net: octeontx2: skip allocating rxvlan for VFs

VFs do not support rxvlan offload yet hence
skip allocation logic.

Signed-off-by: Tomasz Duszynski <tduszynski@marvell.com>
[Kevin: The original patch got from Marvell sdk10.0_19.06]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/net/ethernet/marvell/octeontx2/otx2_pf.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/drivers/net/ethernet/marvell/octeontx2/otx2_pf.c b/drivers/net/ethernet/marvell/octeontx2/otx2_pf.c
index c7a65185bdf3..66cd5967d824 100644
--- a/drivers/net/ethernet/marvell/octeontx2/otx2_pf.c
+++ b/drivers/net/ethernet/marvell/octeontx2/otx2_pf.c
@@ -878,8 +878,9 @@ int otx2_open(struct net_device *netdev)
 	/* Enable link notifications */
 	otx2_cgx_config_linkevents(pf, true);
 
-	/* Alloc rxvlan entry in MCAM */
-	otx2_alloc_rxvlan(pf);
+	/* Alloc rxvlan entry in MCAM for PFs only */
+	if (!(pf->pcifunc & RVU_PFVF_FUNC_MASK))
+		otx2_alloc_rxvlan(pf);
 
 	return 0;
 
-- 
2.17.1

